### PR TITLE
Fixed bug on program admin page

### PR DESCRIPTION
**Pull request by clintonb**
The previous override of save_model() was largely unnecessary and broke because it attempts to set m2m fields on unsaved models. The override now relies on the super method (which performs the correct behavior) and simply does error handling.

The usage of Select2 has been removed from the courses field due to the fact that the widget no longer renders values in the correct order.

ECOM-6238

### CODE REVIEW

**Comment by clintonb** [comment]
@edx/ecommerce please review. I have removed the select2 widget for the courses field. I have spent way too many hours looking for a better solution, and come up empty. The widget no longer works for data that needs to be presented in a set order. This has nothing to do with the backend changes, as the data is stored in the correct order.

File: course_discovery/apps/course_metadata/forms.py [hunk 1]
```diff
@@ -33,15 +32,8 @@ class ProgramAdminForm(HackDjangoAutocompleteMixin, forms.ModelForm):
     class Meta:
         model = Program
         fields = '__all__'
-        from dal import autocomplete
```
Comment (MatthewPiatetsky) [inline course_discovery/apps/course_metadata/forms.py:36]: Will the override for the mixin still work if it is declared after the import? You can try checking out forms.py from master to see if that fixes your issue

Comment (clintonb) [reply course_discovery/apps/course_metadata/forms.py:36]: What override? Did someone implement a hack and not have the professional courtesy to at least leave a comment!?

Comment (MatthewPiatetsky) [reply course_discovery/apps/course_metadata/forms.py:36]: There are multiple comments in the relevant class above.
Agreed that this import should also have a comment

**Review by schenedx — APPROVED** [review]
👍

File: course_discovery/apps/course_metadata/forms.py [hunk 2]
```diff
@@ -8,32 +8,30 @@
...
-
-class ProgramAdminForm(HackDjangoAutocompleteMixin, forms.ModelForm):
+class ProgramAdminForm(forms.ModelForm):
     class Meta:
         model = Program
         fields = '__all__'
-        from dal import autocomplete
+
+        # Monkey patch filter_choices_to_render with our own definition which preserves ordering.
+        autocomplete.ModelSelect2Multiple.filter_choices_to_render = filter_choices_to_render_with_order_preserved
```
Comment (MatthewPiatetsky) [inline course_discovery/apps/course_metadata/forms.py:33]: 👍

**Review by MatthewPiatetsky — APPROVED** [review]

**Pull request merged**